test/blackbox/fips140_authn.bats

```diff
@@ -0,0 +1,217 @@
+
+    # Setup zot server
+    local zot_root_dir=${BATS_FILE_TMPDIR}/zot
+    local zot_config_file=${BATS_FILE_TMPDIR}/zot_config.json
+    ZOT_LOG_FILE=${zot_root_dir}/zot-log.json
+    local zot_htpasswd_file=${BATS_FILE_TMPDIR}/zot_htpasswd
+    zot_port=$(get_free_port_for_service "zot")
+    echo ${zot_port} > ${BATS_FILE_TMPDIR}/zot.port
+    htpasswd -Bbn ${AUTH_USER} ${AUTH_PASS} >> ${zot_htpasswd_file} # bcrypt
+    echo "${AUTH_USER2}:$(openssl passwd -5 ${AUTH_PASS2})" >> ${zot_htpasswd_file} # sha256
+    echo "${AUTH_USER3}:$(openssl passwd -6 ${AUTH_PASS3})" >> ${zot_htpasswd_file} # sha512
+
+    echo ${zot_root_dir} >&3
+
+    mkdir -p ${zot_root_dir}
+
+    touch ${ZOT_LOG_FILE}
+    cat > ${zot_config_file}<<EOF
+{
+  "distSpecVersion":"1.1.1",
+  "storage":{
+    "dedupe": true,
+    "gc": true,
+    "gcDelay": "1h",
+    "gcInterval": "6h",
+    "rootDirectory": "${zot_root_dir}"
+  },
+  "http": {
+		"address": "127.0.0.1",
+		"port": "${zot_port}",
+    "realm":"zot",
+    "auth": {
+      "htpasswd": {
+        "path": "${zot_htpasswd_file}"
+      },
+      "failDelay": 5
+    },
+    "accessControl": {
+      "repositories": {
+        "**": {
+          "anonymousPolicy": ["read"],
+          "defaultPolicy": ["read", "create"]
+        }
+      },
+      "adminPolicy": {
+        "users": ["admin"],
+        "actions": ["read", "create", "update", "delete"]
+      }
+    }
+  },
+  "log":{
+    "level":"debug",
+    "output": "${ZOT_LOG_FILE}"
+  }
+}
+EOF
+    export GODEBUG="fips140=only"
+    zot_serve ${ZOT_PATH} ${zot_config_file}
+    wait_zot_reachable ${zot_port}
+    log_output | jq 'contains("fips140 is currently enabled")?' | grep true
+}
+
+function teardown() {
+    # conditionally printing on failure is possible from teardown but not from from teardown_file
+    cat ${BATS_FILE_TMPDIR}/zot/zot-log.json
+}
+
+function teardown_file() {
+    zot_stop_all
+    unset GODEBUG
+}
+
+@test "push image with regclient" {
+    zot_port=`cat ${BATS_FILE_TMPDIR}/zot.port`
+    run regctl registry set localhost:${zot_port} --tls disabled
+    # bcrypt should fail
+    run regctl registry login localhost:${zot_port} -u ${AUTH_USER} -p ${AUTH_PASS}
+    [ "$status" -eq 0 ]
+    run regctl image copy ocidir://${TEST_DATA_DIR}/busybox:1.36 localhost:${zot_port}/test-regclient
+    [ "$status" -eq 1 ]
+    log_output | jq 'contains("htpasswd bcrypt failed since fips140 is enabled")?' | grep true
+    # sha256 success
+    run regctl registry login localhost:${zot_port} -u ${AUTH_USER2} -p ${AUTH_PASS2}
+    [ "$status" -eq 0 ]
+    run regctl image copy ocidir://${TEST_DATA_DIR}/busybox:1.36 localhost:${zot_port}/test-regclient
+    [ "$status" -eq 0 ]
+    # sha512 success
+    run regctl registry login localhost:${zot_port} -u ${AUTH_USER3} -p ${AUTH_PASS3}
+    [ "$status" -eq 0 ]
+    run regctl image copy ocidir://${TEST_DATA_DIR}/busybox:1.36 localhost:${zot_port}/test-regclient
```

Comment:
Is authentication supposed to be used only in this test? Do we rely on the last logged in user for the others?
Can we logout in teardown_file to ensure we don't impact other tests?